pajbot/managers/connection.py

```diff
@@ -51,13 +51,26 @@ def __init__(self, reactor):
         super().__init__(reactor)
 
         self.num_msgs_sent = 0
+        self.num_whispers_sent_minuite = 0
+        self.num_whispers_sent_second = 0
         self.in_channel = False
 
     def reduce_msgs_sent(self):
         self.num_msgs_sent -= 1
 
-    def can_send(self):
-        return self.num_msgs_sent < TMI.message_limit
+    def reduce_whispers_sent_minuite(self):
+        self.num_whispers_sent_minuite -= 1
+
+    def reduce_whispers_sent_second(self):
+        self.num_whispers_sent_second -= 1
+
+    def can_send(self, whisper=False):
+        if whisper:
+            return self.num_msgs_sent < TMI.message_limit
+        return (
+            self.num_whispers_sent_second < TMI.whispers_message_limit_second
+            and self.num_whispers_sent_minuite < TMI.whispers_message_limit_minute
```

Comment:
typo

Reply:
Additionally, when sending a whisper, num_msgs_sent also counts (the 7500 or 100 are still counted when sending whispers, because they are for all PRIVMSG messages)